lib/index.js

```diff
@@ -125,58 +152,59 @@ class PackageJson {
     return this.#path
   }
 
-  update (content) {
-    // validates both current manifest and content param
-    const invalidContent =
-      typeof this.#manifest !== 'object'
-        || typeof content !== 'object'
-    if (invalidContent) {
-      throw Object.assign(
-        new Error(`Can't update invalid package.json data`),
-        { code: 'EPACKAGEJSONUPDATE' }
-      )
+  get filename () {
+    if (this.path) {
+      return resolve(this.path, 'package.json')
     }
+    return undefined
+  }
 
+  update (content) {
+    if (!this.content) {
+      throw new Error('Can not update without existing data')
+    }
     for (const step of knownSteps) {
-      this.#manifest = step({ content, originalContent: this.#manifest })
+      this.#manifest = step({ content, originalContent: this.content })
     }
 
     // unknown properties will just be overwitten
     for (const [key, value] of Object.entries(content)) {
       if (!knownKeys.has(key)) {
-        this.#manifest[key] = value
+        this.content[key] = value
       }
     }
 
     return this
   }
 
   async save () {
-    if (this.#fromIndex) {
+    if (!this.#canSave) {
       throw new Error('No package.json to save to')
     }
     const {
       [Symbol.for('indent')]: indent,
       [Symbol.for('newline')]: newline,
-    } = this.#manifest
+    } = this.content
 
-    const format = indent === undefined ? '  ' : indent
-    const eol = newline === undefined ? '\n' : newline
+    // These can only be undefined if we didn't parse JSON, which we currently don't support.
+    // const format = indent === undefined ? '  ' : indent
+    // const eol = newline === undefined ? '\n' : newline
```

Comment:
This can be removed.

Reply:
I left this in because I *think* it's only temporary, and we'll need this when we allow for `init` functionality.